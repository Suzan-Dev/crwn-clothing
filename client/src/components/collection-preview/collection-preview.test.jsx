import React from 'react';
import { shallow } from '../../setupTests';
import CollectionPreview from './collection-preview.component';

describe('Testing: CollectionPreview component!', () => {
  const mockTitle = 'Hello World';
  const mockItems = [{ id: 1 }, { id: 2 }];

  const mockProps = {
    title: mockTitle,
    items: mockItems,
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CollectionPreview {...mockProps} />);
  });

  test('should render CollectionPreview component.', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should render title prop correctly.', () => {
    expect(wrapper.find('.title').text()).toBe(mockTitle);
  });

  test('should render num of connected CollectionItem component as in items array.', () => {
    expect(wrapper.find('Connect(CollectionItem)').length).toBe(mockItems.length);
  });
});
