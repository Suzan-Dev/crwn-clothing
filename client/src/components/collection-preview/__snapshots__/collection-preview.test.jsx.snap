// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing: CollectionPreview component! should render CollectionPreview component. 1`] = `
"<div className=\\"collection-preview\\">
  <h1 className=\\"title\\">
    Hello World
  </h1>
  <div className=\\"preview\\">
    <Connect(CollectionItem) item={{...}} />
    <Connect(CollectionItem) item={{...}} />
  </div>
</div>"
`;
